src/borg/remote.py

```diff
@@ -487,8 +487,15 @@ def __init__(self, location, create=False, exclusive=False, lock_wait=None, lock
         self.ignore_responses = set()
         self.responses = {}
         self.ratelimit = SleepingBandwidthLimiter(args.remote_ratelimit * 1024 if args and args.remote_ratelimit else 0)
-
-        self.unpacker = msgpack.Unpacker(use_list=False)
+        # we do not trust the server, use a limited Unpacker to avoid memory DoS via high element counts:
+        self.unpacker = msgpack.Unpacker(use_list=False,
+                                         max_buffer_size=3 * max(BUFSIZE, MAX_OBJECT_SIZE),
+                                         max_str_len=MAX_OBJECT_SIZE, # a chunk or other repo object
+                                         max_bin_len=0,  # not used
+                                         max_array_len=LIST_SCAN_LIMIT,  # repo.list() / .scan() id list result
+                                         max_map_len=100, # misc. result dicts
```

Comment:
not sure about this one either.
i let it crash by using a too low value first and then "rounded up".